test/types/string.js

```diff
@@ -1531,6 +1531,18 @@ describe('string', () => {
             ], { convert: false });
         });
 
+        it('disable existing trim flag when passing enabled: false', () => {
+
+            const schema = Joi.string().trim().trim(false);
```

Comment:
Looks like `.trim()` is repeated twice here.

I also don't see any tests for `.trim(true)` or when the parameter to trim is not a boolean to validate that we are checking the type like other calls (ie. `.trim(42)`).

Last piece on this part would be documentation, this new functionality should be listed in the docs also.

Reply:
Hmm, I just follow the implementation of the `.truncate(enabled?)`

But if it really needed, I'll write some assertion here.